app/models/generic_object_definition.rb

```diff
@@ -49,6 +45,31 @@ def property_defined?(attr)
     property_attribute_defined?(attr) || property_association_defined?(attr) || property_method_defined?(attr)
   end
 
+  def create_object(options)
+    GenericObject.create!({:generic_object_definition => self}.merge(options))
+  end
+
+  # To query based on GenericObject AR attributes and property attributes
+  #   find_objects(:name => "TestLoadBalancer", :uid => '0001', :prop_attr_1 => 10, :prop_attr_2 => true)
+  #
+  # To query based on property associations
+  #   find_objects(:vms => [23, 26])
+  #
+  def find_objects(options)
+    dup = options.stringify_keys!.dup
+    ar_options   = dup.extract!(*(GenericObject.column_names - ["properties"]))
+    json_options = dup.extract!(*(property_attributes.keys + property_associations.keys))
+
+    unless dup.empty?
+      err_msg = _("[%{attrs}]: not defined for Generic Object of %{name}") % {:attrs => dup.keys.join(", "),
```

Comment:
consider the wording `not searchable` instead of `not defined`, for example a method name might be defined but not searchable.